WordPress/Classes/Stores/StatsPeriodStore.swift

```diff
@@ -249,29 +251,37 @@ class StatsPeriodStore: QueryStore<PeriodStoreState, PeriodQuery> {
         processQueries()
     }
 
-    func persistToCoreData() {
-        guard
-            let siteID = SiteStatsInformation.sharedInstance.siteID,
-            let blog = Blog.lookup(withID: siteID, in: ContextManager.shared.mainContext) else {
-                return
+    private func storeDataInCache() {
+        guard let siteID = SiteStatsInformation.sharedInstance.siteID else {
+            return
         }
-
-        _ = state.summary.flatMap { StatsRecord.record(from: $0, for: blog) }
-        _ = state.topPostsAndPages.flatMap { StatsRecord.record(from: $0, for: blog) }
-        _ = state.topReferrers.flatMap { StatsRecord.record(from: $0, for: blog) }
-        _ = state.topClicks.flatMap { StatsRecord.record(from: $0, for: blog) }
-        _ = state.topPublished.flatMap { StatsRecord.record(from: $0, for: blog) }
-        _ = state.topAuthors.flatMap { StatsRecord.record(from: $0, for: blog) }
-        _ = state.topSearchTerms.flatMap { StatsRecord.record(from: $0, for: blog) }
-        _ = state.topCountries.flatMap { StatsRecord.record(from: $0, for: blog) }
-        _ = state.topVideos.flatMap { StatsRecord.record(from: $0, for: blog) }
-        _ = state.topFileDownloads.flatMap { StatsRecord.record(from: $0, for: blog) }
-
-        try? ContextManager.shared.mainContext.save()
```

Comment:
This is the main reason this PR exists.
There are two more remaining `mainContext.save()` calls that I'll address separately.

After the main PRs are merged, I'll remove the Core Data code that manages stats.